agreement_legal/models/res_partner.py

```diff
@@ -23,7 +23,7 @@ def _compute_agreements_count(self):
     def action_open_agreement(self):
         self.ensure_one()
         action = self.env.ref("agreement.agreement_action")
-        result = action.read()[0]
+        result = action.sudo().read()[0]
```

Comment:
A shorter alternative:

```suggestion
        action = self.env["ir.actions.actions"]._for_xml_id("agreement.agreement_action")
```

Reply:
Thank you. I already fixed it.